src/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -58,49 +63,68 @@ internal override async Task<AuthenticationResult> ExecuteAsync(CancellationToke
                     "Token cache is set to null. Silent requests cannot be executed.");
             }
 
-            MsalAccessTokenCacheItem msalAccessTokenItem = null;
-
-            // Look for access token
-            if (!ForceRefresh)
+            if (!String.IsNullOrWhiteSpace(UserProvidedRefreshToken))
             {
-                msalAccessTokenItem =
-                    await TokenCache.FindAccessTokenAsync(AuthenticationRequestParameters).ConfigureAwait(false);
+                AuthenticationRequestParameters.RequestContext.Logger.Verbose("Exchanging access token...");
+                await ResolveAuthorityEndpointsAsync().ConfigureAwait(false);
+                var msalTokenResponse = await SendTokenRequestAsync(GetBodyParameters(UserProvidedRefreshToken), cancellationToken)
+                                            .ConfigureAwait(false);
+
+                if (msalTokenResponse.RefreshToken == null)
```

Comment:
wouldn't this be a problem?  Should we log this as a warning/error?  Why would we not get a refresh token here?